app/tests/unit/test_gw2_crawler_client.py

```
@@ -0,0 +1,436 @@
+
+                    assert result == [1, 2, 3]
+                    assert call_count == 3
+                    # Should have slept twice (after first two failures)
+                    assert sleep_count == 2
+
+    @pytest.mark.asyncio
+    async def test_retries_on_timeout(self, mock_http_client, mock_http_response):
+        """Test that crawler retries on timeout errors."""
+        call_count = 0
+
+        async def mock_wait(*args, **kwargs):
```

Ruff (0.14.4)

Unused function argument: `args`

(ARG001)